string.go

```diff
@@ -28,25 +27,30 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
-func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
+// AESGCMEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESGCMEncrypt(key, plaintext []byte) ([]byte, error) {
 	block, err := aes.NewCipher(key)
 	if err != nil {
 		return nil, err
 	}
 
-	b := base64.StdEncoding.EncodeToString(plaintext)
 	gcm, err := cipher.NewGCM(block)
 	if err != nil {
 		return nil, err
 	}
 
-	ciphertext := gcm.Seal(nil, nonce, []byte(b), nil)
-	return ciphertext, nil
+	nonce := make([]byte, gcm.NonceSize())
+	_, err = rand.Read(nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	ciphertext := gcm.Seal(nil, nonce, plaintext, nil)
+	return append(ciphertext, nonce...), nil
```

Comment:
If we store nonce in ciphertext... doesn't the attacker can just take and use it?